ambry-clustermap/src/main/java/com.github.ambry.clustermap/CloudServiceReplica.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+import com.github.ambry.config.ClusterMapConfig;
+import org.json.JSONObject;
+
+import static com.github.ambry.clustermap.ClusterMapSnapshotConstants.*;
+
+
+/**
+ * An implementation of {@link AmbryReplica} that is meant to route requests towards cloud destinations using a standard
+ * client library. The difference between this and {@link CloudReplica} is that {@link CloudReplica} is meant to
+ * represent a VCR host instead of just being used to route requests towards a local client library. Additionally,
+ * implementing {@link AmbryReplica} currently allows for easier interop with the rest of the helix cluster management
+ * codebase.
```

Comment:
Thanks for explaining the difference. So, `ClouldServiceReplica` is used by frontend for now and primarily used to route requests to cloud, right? For server, it interacts with `CloudReplica` to perform two-way replication. (Correct me if I have misunderstood)

Reply:
Since `CloudServiceReplica` will be in HelixClusterManager, so server is able to see it. I guess there is no server logic directly depends on `CloudServiceReplica`.

Reply:
Yes it is for frontends, but it would only be present for the nodes that are configured to see it, using `clusterMapDcsZkConnectStrings`. Frontends can be configured to include it, and servers could be configured without it.

Yes a `CloudReplica` currently represents a replica that is present on a VCR host. In the long run, I would hope to also fold this one into the regular clustermap as a `VcrReplica`. I think that would help to clean up our 2-way replication story and allow us to share more code between the many ReplicationEngine variants that currently exist.